### PR TITLE
Add HTTP processor to client.

### PR DESCRIPTION
**Pull request by omersalman17**
Add HTTP processor to Artifactory client.

I have read the CLA Document and I hereby sign the CLA.

### CODE REVIEW

**Comment by eyalbe4** [comment]
Thanks for contributing this @omersalman17!
It'll be nice if on future contributions, you'll be following the [Contribution Guidelines](https://github.com/jfrog/artifactory-client-java#contributing-code) which state that *Pull requests should be created on the dev branch.*

File: services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java [hunk 1]
```diff
@@ -156,6 +164,10 @@ private CloseableHttpClient createClientBuilder(URI uri) {
         for (HttpRequestInterceptor httpRequestInterceptor : requestInterceptorList) {
             artifactoryHttpClient.addInterceptorLast(httpRequestInterceptor);
         }
+
+        artifactoryHttpClient.addRequestInterceptor(httpProcessor);
+        artifactoryHttpClient.addResponseInterceptor(httpProcessor);
```
Comment (eyalbe4) [inline services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java:169]: Can you please share some information about the above two lines?
What's the goal of adding the ```httpProcessor``` as request and response interceptors here?
You already exposed an API that adds the interceptors below, so why do you force setting them here inside the builder? 
image

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java (hunk 1 above)
Comment (omersalman17) [reply services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java:169]: The above two lines are the ones that actually assign the processor to the internal ArtifactoryHttpClient object which is used to built the client, all of this happens internally on  createClientBuilder(URI uri) method with no access from outside, hence removing the two of them will resolve that the processor won't be assigned to the client upon creation (please take a look at the complete implementation of the method and how it is used for the client creation).
Thank you 😊 .

File: httpClient/src/main/java/org/jfrog/artifactory/client/httpClient/http/HttpBuilderBase.java [hunk 2]
```diff
@@ -194,6 +194,16 @@ public T authentication(String username, String password, boolean allowAnyHost)
         return self();
     }
 
+    public T addRequestInterceptor(HttpRequestInterceptor interceptor) {
```
[the remark this answers is not in the document]
Comment (omersalman17) [reply httpClient/src/main/java/org/jfrog/artifactory/client/httpClient/http/HttpBuilderBase.java:197]: 👍🏻

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**